src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -84,20 +84,35 @@ public override void Inject<T>(PropagationContext context, T carrier, Action<T,
         if (e.MoveNext() == true)
         {
             int itemCount = 0;
-            StringBuilder baggage = new StringBuilder();
+            StringBuilder baggage = null;
             do
             {
                 KeyValuePair<string, string> item = e.Current;
-                if (string.IsNullOrEmpty(item.Value))
+
+                if (!ValidateKey(item.Key))
                 {
                     continue;
                 }
 
-                baggage.Append(Uri.EscapeDataString(item.Key)).Append('=').Append(Uri.EscapeDataString(item.Value)).Append(',');
+                if (baggage == null)
```

Comment:
Is there a need to differentiate null vs empty? This would get checked in every iteration of the loop.

Reply:
There isn't a need to differentiate between these, I will modify it.